### ISSUE TITLE
Wrong type argument: integer-or-marker-p, nil

### ISSUE THREAD

**Issue by otech-nl** [org-roam/org-roam#2361]
### Description

I use org-roam on Linux and Windows, and recently added MacOS to the mix. The configuration that worked for me on other platforms now gives a "Wrong type argument: integer-or-marker-p, nil" when the org-roam db is updated.

#### Steps to Reproduce

The relevant part of my config (with [straight use-package](https://github.com/radian-software/straight.el#integration-with-use-package)):
```
  (use-package org-roam
    :ensure t
    :init
    (setq org-roam-v2-ack t)
    :after org
)
```

The error occurs when I start Emacs, but I can also trigger it with `org-roam-db-sync`.

I also tried to remove the org-roam database and the complete `straight` folder before starting Emacs to force a clean start, but this gives the same result.

#### Backtrace

I don't get a backtrace, even with `debug-on-error t` (or `--debug-init`). Instead I get the following message for each of my `org-roam` files:
```
Error (org-roam): Failed to process <filename>.org with error Wrong type argument: integer-or-marker-p, nil, skipping... Disable showing Disable logging
```
This error messages comes from  [org-roam-db.el:698](https://github.com/org-roam/org-roam/blob/5c06471c3a11348342719fd9011486455adeb701/org-roam-db.el#L698), which is confirmed because I see the message "Processing modified files..."

#### Expected Results

Updated database without errors

#### Actual Results

Error messages as described above.

The database _does_ get filled with tables and data. I can't say if its complete.

### Environment

Emacs installed with homebrew cask, which uses https://emacsformacosx.com/.

- Emacs: GNU Emacs 28.2 (build 1, aarch64-apple-darwin21.1.0, NS appkit-2113.00 Version 12.0.1 (Build 21A559))
 of 2023-02-22
- Framework: N/A
- Org: Org mode version 9.7-pre (release_9.6.7-543-g501be3 @ /Users/steets/.emacs.d/straight/build/org/)
- Org-roam: v2.2.2-42-g5c06471
- sqlite-connector: sqlite

**Comment by max6166** [comment]
+1. Exactly the same problem. Using the latest org-mode 9.7pre from 2023-07-07 and org-roam 2.2.2.

The only clue I've been able to find is related to external links. Both of the files that cause this error contain external links to web pages. Deleting those links solves the problem. 

So far, it seems the problem can potentially be caused by any link, but sometimes the same link won't cause this issue.

**Comment by otech-nl** [comment]
I checked a subset of my notes that got successfully imported and can confirm they did not contain links.

**Comment by vderyagin** [comment]
Also had this, slightly downgrading org-mode (to commit `9082fa889`) resolved the issue, didn't investigate further than that

**Comment by max6166** [comment]
Below is a simple node example that triggers the error. There are no square brackets around the link info. This is the complete text of the node.

```
:PROPERTIES:
:ID:       C8AD00C6-A447-4BA4-B844-E48262019C1B
:END:
#+title: Test

id:D6765314-53B6-4D90-A0F7-889154A4C361
```

Removing the colon from the last line fixes the problem. 

```
:PROPERTIES:
:ID:       C8AD00C6-A447-4BA4-B844-E48262019C1B
:END:
#+title: Test

idD6765314-53B6-4D90-A0F7-889154A4C361
```

**Comment by ifinkelstein** [comment]
Quick note that `M-x straight-freeze-versions` should be run first to write the `.emacs.d/straight/versions/default.el` file. This will lock down all package versions, unless the file is edited to remove all but the `org`-related commit.


> > Also had this, slightly downgrading org-mode (to commit `9082fa889`) resolved the issue, didn't investigate further than that
> 
> Thanks! With `straight.el` I can [pin my version](https://github.com/radian-software/straight.el#lockfile-management) in `.emacs.d/straight/versions/default.el` to [org@9.5.5](https://github.com/bzg/org-mode/commit/8ef6205a560cd3a92f8c5f8fe34953b80121c2cb) with:
> 
> ```
> (("org" . "8ef6205a560cd3a92f8c5f8fe34953b80121c2cb"))
> :gamma
> ```
> 
> This gets rid of the error for me.
> 
> If I understand correctly, straight installs the latest version by default (`org@9.7-pre`). The above suggests that this pre-release breaks `org-roam`. I am not sure if/how to inform the maintainers of `org`/`org-roam`. Any suggestions?

**Comment by Itrekr** [comment]
I also have this issue. Any note with links fully messes up my emacs somehow.

**Comment by jmburgos** [comment]
+1 Same problem here.  I am using:

Org roam: v2.2.2-42-g5c06471
Org mode version 9.7-pre (release_9.6.7-572-g39de4a)
GNU Emacs 30.0.50 (build 2, x86_64-pc-linux-gnu, GTK+ Version 3.24.37, cairo version 1.17.8) of 2023-05-09.

So if using the pre-release version of org mode is the problem, how do I exactly reverse to a previous version?  I am using straight, but I my expertise is very thin.  From what was described before, I need to:

1. Run straight-freeze-versions to generate the default.el file.
2. Edit the org entry in the file

... and then?  I am not sure how to force straight to re-install org mode.  Could you give me some points? Thanks! :)

**Comment by Itrekr** [comment]
> +1 Same problem here.  I am using:
> 
> Org roam: v2.2.2-42-g5c06471
> Org mode version 9.7-pre (release_9.6.7-572-g39de4a)
> GNU Emacs 30.0.50 (build 2, x86_64-pc-linux-gnu, GTK+ Version 3.24.37, cairo version 1.17.8) of 2023-05-09.
> 
> So if using the pre-release version of org mode is the problem, how do I exactly reverse to a previous version?  I am using straight, but I my expertise is very thin.  From what was described before, I need to:
> 
> 1. Run straight-freeze-versions to generate the default.el file.
> 2. Edit the org entry in the file
> 
> ... and then?  I am not sure how to force straight to re-install org mode.  Could you give me some points? Thanks! :)

I second this. If someone could explain how to get this done for the novices like me/us I'd be very grateful. My whole workflow is broken now so I'd love to fix it.

**Comment by otech-nl** [comment]
Check my [previous reply](https://github.com/org-roam/org-roam/issues/2361#issuecomment-1632943836) for configuring `straight.el`.

**Comment by jmburgos** [comment]
@otech-nl, I was trying to follow your instructions.  I ran straight-freeze-versions and editec the defaul.el file as you did.  But I am not sure what to do after that.  I did straight-pull org and straight-merge org, but my org mode version is still 9.7-pre.  As I said, my expertise is thin... :)

**Comment by otech-nl** [comment]
> @otech-nl, I was trying to follow your instructions. I ran `straight-freeze-versions` and edited the `defaul.el `file as you did. But I am not sure what to do after that. I did straight-pull org and straight-merge org, but my org mode version is still 9.7-pre. As I said, my expertise is thin... :)

You may have to delete `org` from `.emacs.d/straight/repos` and/or `.emacs.d/straight/build` and restart Emacs

> An of course, this happens when I on a deadline and I need to use org-roam a lot. Murphy's law, I guess...

More like Finagle's corollary ;)
Good luck!

**Comment by draxil** [comment]
Is there a bug report for this in org-mode?

**Comment by jmburgos** [comment]
Thank you @otech-nl, deleting org from those locations was the key, and in hindsight something obvious to do.  Now I am in org 9.5.5. and everything works fine.  Hopefully this will get sorted soon.

**Comment by Itrekr** [comment]
> Check my [previous reply](https://github.com/org-roam/org-roam/issues/2361#issuecomment-1632943836) for configuring `straight.el`.

I can't get straight.el to work. I have tried using straight-freeze-versions but this gives me a vague error that "Caches are still outdated; something is seriously wrong" but I can't find the cause of this. Am I missing some glaringly obvious step in the straight.el process?

**Comment by goofansu** [comment]
I confirm the latest stable version [9.6.7](https://git.savannah.gnu.org/cgit/emacs/org-mode.git/tag/?h=release_9.6.7) also works. Configure in `packages.el` if using Doom Emacs:

```shell
(package! org :pin "5890ac")
```

**Comment by dustinfarris** [comment]
> I confirm the latest stable version [9.6.7](https://git.savannah.gnu.org/cgit/emacs/org-mode.git/tag/?h=release_9.6.7) also works. Configure in `packages.el` if using Doom Emacs:
> 
> ```shell
> (package! org :pin "5890ac")
> ```

Thanks for the tip, but where did you get that commit?  I couldn't find it anywhere.

I pegged to 9.6.7 with:

```elisp
(package! org :pin "ca873f7")
```

and things are working much smoother now :-)

**Comment by goofansu** [comment]
@dustinfarris In the link: https://git.savannah.gnu.org/cgit/emacs/org-mode.git/tag/?h=release_9.6.7

**Comment by yantar92** [comment]
See my comment in https://github.com/radian-software/straight.el/issues/1107 for the likely reason.
I believe that the issue belongs to straight.

**Comment by yanboyang713** [comment]
Hello @dustinfarris ,

I pined to "ca873f7" and "5890ac", but still does not work.
Which version of emacs are you using? I am using 28.2.
Thanks

**Comment by yantar92** [comment]
@yanboyang713 You may need to re-compile everything that depends on Org. Because the issue is related to inlined functions that may be left in mixed state if you switched back-and-forth between the pins.

**Comment by yanboyang713** [comment]
Thanks @yantar92 . Working now.

**Comment by goofansu** [comment]
> I pegged to 9.6.7 with:
> 
> ```emacs-lisp
> (package! org :pin "ca873f7")
> ```

Hi @dustinfarris, where did you get the `ca873f7` commit? I found `5890ac` is not usable when `doom upgrade`.

**Comment by dustinfarris** [comment]
5890ac is the tag ref.  The actual commit that was tagged is ca873f7https://git.savannah.gnu.org/cgit/emacs/org-mode.git/commit/?h=release_9.6.7&id=ca873f7fe47546bca19821f1578a6ab95bf5351cDustinSent from my iPhoneOn Aug 7, 2023, at 19:27, Yejun Su ***@***.***> wrote:﻿

I pegged to 9.6.7 with:
(package! org :pin "ca873f7")

Hi @dustinfarris, where did you get the ca873f7 commit? I found 5890ac is not usable when doom upgrade.

**Comment by goofansu** [comment]
@dustinfarris Got it, thank you!

**Comment by progfolio** [comment]
Should be fixed on develop branch of straight.el.
Testing is appreciated.

**Comment by Samoed** [comment]
@progfolio For me, everything is working at the latest commit

**Comment by progfolio** [comment]
> I am still having the issue with release_9.6.7-644-gf80c82. :(

What's the output of `M-x straight-version`?

**Comment by jmburgos** [comment]
` (HEAD -> master, origin/master, origin/HEAD) ff63b15 2023-06-15`

**Comment by progfolio** [comment]
> ` (HEAD -> master, origin/master, origin/HEAD) ff63b15 2023-06-15`

You would need to update straight's "develop" branch.
e.g.

```emacs-lisp
(setq straight-repository-branch "develop")
```
Before straight's bootstrapping snippet.

**Comment by jmburgos** [comment]
Ah! You are right @progfolio.  Now things are working!

**Comment by Samoed** [comment]
@gicrisf You can pin org-mode version in `doom/packages.el`. And after that delete org repos and then `doom sync`.

**Comment by gicrisf** [comment]
@Samoed Does it work too?! This is great, I thought the modules would be loaded first

**Comment by Enemoy** [comment]
I pinned it according to @gicrisf but I still have the same error in files that contain links. The error messages start to stack up. I can still use org mode but every time I make an update I have the same error coming up.

Do I have to pin the version in `packages.el` or where do I have to pin it? And which version of the package should I pin?

**Comment by jacmoe** [comment]
@Enemoy : 

Since you are using Doom Emacs, you are using straight and other supporting packages . . . 

Pin Org in packages.el ->

`;; -*- no-byte-compile: t; -*-
;;; $DOOMDIR/packages.el

;; Fix for org-roam link issue
(package! org :pin "ca873f7")
`

Then remove the `.local` directory in the main Doom dir.

Then simply run `doom sync` - since you removed the local dir, Doom should automatically download and install all the packages.

It should all complete without errors.

Now, you can safely and successfully perform `doom upgrade`

**Comment by 64J0** [comment]
+1 facing this problem as well.

**Comment by AlecVercruysse** [comment]
@progfolio Thank you for looking into this, moving to the development branch of straight.el fixes this problem for me.

**Comment by zot** [comment]
For anyone using `elpaca`, this `use-package` clause works for me (maybe there's a better one, beats me):

```elisp
(use-package org
  ;; pin org at 9.6 for now
  ;; modified from elpaca/cache/gnu-elpa.eld to use the release_9.6 branch
  :elpaca (:package "org" :repo "git://git.sv.gnu.org/emacs/elpa" :local-repo "org" :branch "release_9.6" :files ("*" (:exclude ".git" "etc/schema" "contrib" "mk" "testing" "lisp/org-install.el"))))
```

**Comment by progfolio** [comment]
> For anyone using `elpaca`, this `use-package` clause works for me (maybe there's a better one, beats me):
> 
> ```emacs-lisp
> (use-package org
>   ;; pin org at 9.6 for now
>   ;; modified from elpaca/cache/gnu-elpa.eld to use the release_9.6 branch
>   :elpaca (:package "org" :repo "git://git.sv.gnu.org/emacs/elpa" :local-repo "org" :branch "release_9.6" :files ("*" (:exclude ".git" "etc/schema" "contrib" "mk" "testing" "lisp/org-install.el"))))
> ```

Shouldn't need the workaround if you are using Elpaca. See the test case below:


<details open><summary>Test Case</summary>

[How to run this test?](https://github.com/progfolio/elpaca/wiki/Troubleshooting#the-elpaca-test-macro)

```emacs-lisp
(elpaca-test
  :early-init (setq elpaca-menu-functions '(elpaca-menu-org elpaca-menu-melpa))
  :init (elpaca (compat :files ("*" (:exclude ".git")) :local-repo "compat" :repo
                        "https://github.com/emacs-compat/compat"))
  (elpaca org)
  (elpaca org-roam
    (setq org-roam-directory "/tmp/test/" org-roam-v2-ack t))
  (elpaca-wait)
  (org-roam-db-sync))
```

</details>
<details><summary>Host Env</summary>

<table>
<tr><td>elpaca</td><td>1008684 HEAD -> feat/version-checks</td>
<tr><td>isntaller</td><td>0.6</td>
<tr><td>emacs</td><td>GNU Emacs 30.0.50 (build 1, x86_64-pc-linux-gnu, GTK+ Version 3.24.38, cairo version 1.18.0) of 2023-12-06</td>
<tr><td>git</td><td>git version 2.43.0</td>
</table>
</details>

<details><summary>Output</summary>

```emacs-lisp
  INFO     Scraping files for loaddefs... 
  INFO     Scraping files for loaddefs...done
  GEN      ../elpaca-autoloads.el
Cloning into '/tmp/elpaca.wsMEx5/elpaca/repos/elpaca'...
Your branch is up to date with 'origin/master'.
Checking /tmp/elpaca.wsMEx5/elpaca/repos/elpaca...
Compiling /tmp/elpaca.wsMEx5/elpaca/repos/elpaca/elpaca-info.el...
Compiling /tmp/elpaca.wsMEx5/elpaca/repos/elpaca/elpaca-log.el...
Compiling /tmp/elpaca.wsMEx5/elpaca/repos/elpaca/elpaca-manager.el...
Compiling /tmp/elpaca.wsMEx5/elpaca/repos/elpaca/elpaca-menu-elpa.el...
Compiling /tmp/elpaca.wsMEx5/elpaca/repos/elpaca/elpaca-menu-melpa.el...
Compiling /tmp/elpaca.wsMEx5/elpaca/repos/elpaca/elpaca-menu-org.el...
Compiling /tmp/elpaca.wsMEx5/elpaca/repos/elpaca/elpaca-process.el...
Compiling /tmp/elpaca.wsMEx5/elpaca/repos/elpaca/elpaca-test.el...
Compiling /tmp/elpaca.wsMEx5/elpaca/repos/elpaca/elpaca-ui.el...
Compiling /tmp/elpaca.wsMEx5/elpaca/repos/elpaca/elpaca.el...
Checking /tmp/elpaca.wsMEx5/elpaca/repos/elpaca/doc...
Compiling /tmp/elpaca.wsMEx5/elpaca/repos/elpaca/doc/early-init.el...
Compiling /tmp/elpaca.wsMEx5/elpaca/repos/elpaca/doc/init.el...
Checking /tmp/elpaca.wsMEx5/elpaca/repos/elpaca/extensions...
Compiling /tmp/elpaca.wsMEx5/elpaca/repos/elpaca/extensions/elpaca-use-package.el...
Checking /tmp/elpaca.wsMEx5/elpaca/repos/elpaca/images...
Checking /tmp/elpaca.wsMEx5/elpaca/repos/elpaca/test...
Compiling /tmp/elpaca.wsMEx5/elpaca/repos/elpaca/test/elpaca-test.el...
Done (Total of 11 files compiled, 3 skipped in 4 directories)

Downloading MELPA recipes...
Downloading MELPA recipes...100%
Clearing removed files... 
Clearing removed files...done
Processing modified files... 
Processing modified files...4% 
Unable to read file "/home/n/git/Emacs/org-html-themes/setup/theme-readtheorg.setup"
Unable to read file "/home/n/git/Emacs/org-html-themes/setup/theme-readtheorg.setup"
Unable to read file "/home/n/git/Emacs/org-html-themes/setup/theme-readtheorg.setup"
Unable to read file "/home/n/git/Emacs/org-html-themes/setup/theme-readtheorg.setup"
Processing modified files...11% 
Processing modified files...12% 
Processing modified files...13% 
Processing modified files...17% 
Processing modified files...24% 
Processing modified files...31% 
Processing modified files...32% 
Processing modified files...37% 
Processing modified files...44% 
Processing modified files...45% 
Processing modified files...46% 
Processing modified files...47% 
Processing modified files...48% 
Processing modified files...49% 
Processing modified files...50% 
Processing modified files...51% 
Processing modified files...52% 
Processing modified files...53% 
Processing modified files...54% 
Processing modified files...61% 
Processing modified files...77% 
Processing modified files...90% 
Processing modified files...91% 
Processing modified files...92% 
Processing modified files...93% 
Processing modified files...94% 
Processing modified files...99% 
Processing modified files...done

 Test Env

Elpaca ad60abc HEAD -> master, origin/master, origin/HEAD
installer:      0.6
emacs-version:  GNU Emacs 30.0.50 (build 1, x86_64-pc-linux-gnu, GTK+ Version 3.24.38, cairo version 1.18.0)
 of 2023-12-06
git --version:  git version 2.43.0
```
</details>

**Comment by zot** [comment]
Huh, well it was loading 0rg 9.7.X and breaking org-roam until I fixed it with that just a few days ago. So at least one person had the problem (me).

**Comment by progfolio** [comment]
> Huh, well it was loading 0rg 9.7.X and breaking org-roam until I fixed it with that just a few days ago. So at least one person had the problem (me).

Here's another test, explicitly listing the version of Org (9.7-pre):


<details open><summary>Test Case</summary>

[How to run this test?](https://github.com/progfolio/elpaca/wiki/Troubleshooting#the-elpaca-test-macro)

```emacs-lisp
(elpaca-test
  :early-init (setq elpaca-menu-functions '(elpaca-menu-org elpaca-menu-melpa))
  :init (elpaca (compat :files ("*" (:exclude ".git")) :local-repo "compat" :repo
                        "https://github.com/emacs-compat/compat"))
  (elpaca org)
  (elpaca org-roam
    (setq org-roam-directory "/tmp/test/" org-roam-v2-ack t))
  (elpaca-wait)
  (org-roam-db-sync)
  (org-version nil 'full 'message))
```

</details>
<details><summary>Host Env</summary>

<table>
<tr><td>elpaca</td><td>1008684 HEAD -> feat/version-checks</td>
<tr><td>isntaller</td><td>0.6</td>
<tr><td>emacs</td><td>GNU Emacs 30.0.50 (build 1, x86_64-pc-linux-gnu, GTK+ Version 3.24.38, cairo version 1.18.0) of 2023-12-06</td>
<tr><td>git</td><td>git version 2.43.0</td>
</table>
</details>

<details><summary>Output</summary>

```emacs-lisp
  INFO     Scraping files for loaddefs... 
  INFO     Scraping files for loaddefs...done
  GEN      ../elpaca-autoloads.el
Cloning into '/tmp/elpaca.KZ2kgv/elpaca/repos/elpaca'...
Your branch is up to date with 'origin/master'.
Checking /tmp/elpaca.KZ2kgv/elpaca/repos/elpaca...
Compiling /tmp/elpaca.KZ2kgv/elpaca/repos/elpaca/elpaca-info.el...
Compiling /tmp/elpaca.KZ2kgv/elpaca/repos/elpaca/elpaca-log.el...
Compiling /tmp/elpaca.KZ2kgv/elpaca/repos/elpaca/elpaca-manager.el...
Compiling /tmp/elpaca.KZ2kgv/elpaca/repos/elpaca/elpaca-menu-elpa.el...
Compiling /tmp/elpaca.KZ2kgv/elpaca/repos/elpaca/elpaca-menu-melpa.el...
Compiling /tmp/elpaca.KZ2kgv/elpaca/repos/elpaca/elpaca-menu-org.el...
Compiling /tmp/elpaca.KZ2kgv/elpaca/repos/elpaca/elpaca-process.el...
Compiling /tmp/elpaca.KZ2kgv/elpaca/repos/elpaca/elpaca-test.el...
Compiling /tmp/elpaca.KZ2kgv/elpaca/repos/elpaca/elpaca-ui.el...
Compiling /tmp/elpaca.KZ2kgv/elpaca/repos/elpaca/elpaca.el...
Checking /tmp/elpaca.KZ2kgv/elpaca/repos/elpaca/doc...
Compiling /tmp/elpaca.KZ2kgv/elpaca/repos/elpaca/doc/early-init.el...
Compiling /tmp/elpaca.KZ2kgv/elpaca/repos/elpaca/doc/init.el...
Checking /tmp/elpaca.KZ2kgv/elpaca/repos/elpaca/extensions...
Compiling /tmp/elpaca.KZ2kgv/elpaca/repos/elpaca/extensions/elpaca-use-package.el...
Checking /tmp/elpaca.KZ2kgv/elpaca/repos/elpaca/images...
Checking /tmp/elpaca.KZ2kgv/elpaca/repos/elpaca/test...
Compiling /tmp/elpaca.KZ2kgv/elpaca/repos/elpaca/test/elpaca-test.el...
Done (Total of 11 files compiled, 3 skipped in 4 directories)

Downloading MELPA recipes...
Downloading MELPA recipes...100%
Clearing removed files... 
Clearing removed files...done
Processing modified files... 
Processing modified files...4% 
Unable to read file "/home/n/git/Emacs/org-html-themes/setup/theme-readtheorg.setup"
Unable to read file "/home/n/git/Emacs/org-html-themes/setup/theme-readtheorg.setup"
Unable to read file "/home/n/git/Emacs/org-html-themes/setup/theme-readtheorg.setup"
Unable to read file "/home/n/git/Emacs/org-html-themes/setup/theme-readtheorg.setup"
Processing modified files...11% 
Processing modified files...12% 
Processing modified files...13% 
Processing modified files...17% 
Processing modified files...24% 
Processing modified files...28% 
Processing modified files...32% 
Processing modified files...33% 
Processing modified files...38% 
Processing modified files...44% 
Processing modified files...45% 
Processing modified files...46% 
Processing modified files...47% 
Processing modified files...48% 
Processing modified files...49% 
Processing modified files...50% 
Processing modified files...51% 
Processing modified files...52% 
Processing modified files...53% 
Processing modified files...54% 
Processing modified files...55% 
Processing modified files...59% 
Processing modified files...64% 
Processing modified files...87% 
Processing modified files...90% 
Processing modified files...91% 
Processing modified files...92% 
Processing modified files...93% 
Processing modified files...94% 
Processing modified files...97% 
Processing modified files...done
Org mode version 9.7-pre (9.7-pre-n/a-gb8d27b
 @ /tmp/elpaca.KZ2kgv/elpaca/builds/org/)

 Test Env

Elpaca ad60abc HEAD -> master, origin/master, origin/HEAD
installer:      0.6
emacs-version:  GNU Emacs 30.0.50 (build 1, x86_64-pc-linux-gnu, GTK+ Version 3.24.38, cairo version 1.18.0)
 of 2023-12-06
git --version:  git version 2.43.0
```
</details>


Do you recall what steps you took to see the error?
Were packages updated and then the error started happening, fresh install, etc?

**Comment by kentbull** [comment]
Straight package manage user here.

Pinning to "ca873f7" worked for me with the following config:  
```elisp
(use-package org-roam
  :straight (org-roam :type git :host github :repo "org-roam/org-roam" :commit "ca873f7")
  :init
  (setq org-roam-v2-ack t)
  :after org
  :custom
  (org-roam-directory "~/org/roam-brain")
  (org-roam-completion-everywhere t)
  :bind (("C-c n l" . org-roam-buffer-toggle)
         ("C-c n f" . org-roam-node-find)
         ("C-c n i" . org-roam-node-insert)
         :map org-mode-map
         ("C-M-i" . completion-at-point))
  :config
  (org-roam-setup))
```
Once you pin the version then you need to restart Emacs.

**Comment by progfolio** [comment]
Kent Bull <notifications@github.com> writes:

>   :straight (org-roam :type git :host github :repo "org-roam/org-roam" :commit "ca873f7")
> Once you pin the version then you need to restart Emacs.

FYI, straight.el has no :commit recipe keyword.

From the README:

> We support :branch, but not :commit or :version-regexp. To lock a
> package to a specific commit, use a lockfile. See also #246 for
> discussion of extensions to the recipe to support package pinning, which
> is a planned feature.

You'll want to use a [lockfile](https://github.com/radian-software/straight.el?tab=readme-ov-file#lockfile-management).

For [Elpaca](https://github.com/progfolio/elpaca) users, lockfiles are a planned feature.
However, Elpaca does have more flexible recipe keywords such as `:ref` and `:pin`.

**Comment by kentbull** [comment]
Ok, thanks for the correction. You're right. This means I don't know why it works for me.

**Comment by lum7na** [comment]
I've identified a temporary fix if you're using Org 9.7.

https://github.com/org-roam/org-roam/blob/8667e441876cd2583fbf7282a65796ea149f0e5f/org-roam-db.el#L572

Due to modifications in the Org Element API, the `begin` property was removed, leading to the aforementioned error. Replacing `(org-element-property :begin link))` with `(aref (org-element-property :standard-properties link) 0)` can resolve this issue.

**Comment by yantar92** [comment]
lum7na ***@***.***> writes:

> I've identified a temporary fix if you're using Org 9.7.
>
> https://github.com/org-roam/org-roam/blob/8667e441876cd2583fbf7282a65796ea149f0e5f/org-roam-db.el#L572
>
> Due to modifications in the Org Element API, the `begin` property was
> removed, leading to the aforementioned error.

This is not true. :begin property is still there, but the implementation
details of the org-element object structure have changed.

> Replacing `(org-element-property :begin link))` with `(aref (org-element-property :standard-properties link) 0)` can resolve this issue.

I advice against using this implementation detail.
(org-element-property :begin link)) still works. But you may have to
make sure that the inlined functions are re-compiled. After upgrading
Org mode, all the users (including org-roam) of Org element API must be
re-compiled.

-- 
Ihor Radchenko // yantar92,
Org mode contributor,
Learn more about Org mode at <https://orgmode.org/>.
Support Org development at <https://liberapay.com/org-mode>,
or support my work at <https://liberapay.com/yantar92>

**Comment by cbkerr** [comment]
Just a ping for anyone who pinned to an older version of org, you'll want to [update to org 9.7.5 for a security vulnerability fix](https://list.orgmode.org/87sex5gdqc.fsf@localhost/T/#u).

Org-roam from straight.el worked for me after I deleted the `straight/build` directory to trigger a rebuild of everything.

**Comment by bff12** [comment]
I had the same problem but it's because I didn't rebuild all the packages when migrating to emacs-plus 30 on macOS. `(package-recompile-all)` can help anyone who has the same problem.

**Comment by borwickatuw** [comment]
FYI `(package-recompile-all)` addressed the errors for me.

**Comment by rolandog** [comment]
I'm not able to make `package-recompile-all` work as a solution to this problem for me.  It may be due to the fact that I'm using Guix on a "foreign distro" (as a package manager, for the moment):

```
Debugger entered--Lisp error: (file-error "Removing old name" "Read-only file system" "/gnu/store/1z4naashrr0avlbcb22ix5l61jbap83v-emacs-...")
  #<subr delete-file>("/gnu/store/1z4naashrr0avlbcb22ix5l61jbap83v-emacs-..." nil)
  apply(#<subr delete-file> ("/gnu/store/1z4naashrr0avlbcb22ix5l61jbap83v-emacs-..." nil))
  delete-file("/gnu/store/1z4naashrr0avlbcb22ix5l61jbap83v-emacs-..." nil)
  package-recompile(#s(package-desc :name deferred :version (0 5 1) :summary "Simple asynchronous functions for emacs lisp" :reqs ((emacs (24 4))) :kind nil :archive nil :dir "/gnu/store/1z4naashrr0avlbcb22ix5l61jbap83v-emacs-..." :extras ((:url . "https://github.com/kiwanami/emacs-deferred") (:keywords "deferred" "async") (:maintainer "SAKURAI Masashi" . "m.sakuraiatkiwanami.net") (:authors ("SAKURAI Masashi" . "m.sakuraiatkiwanami.net"))) :signed nil))
  package-recompile-all()
  funcall-interactively(package-recompile-all)
  command-execute(package-recompile-all record)
  execute-extended-command(nil "package-recompile-all" nil)
  funcall-interactively(execute-extended-command nil "package-recompile-all" nil)
  command-execute(execute-extended-command)
```

I also tried interactively recompiling only org-roam (`package-recompile`), which resulted in:

```
Debugger entered--Lisp error: (file-error "Removing old name" "Read-only file system" "/gnu/store/61zhv116kl3waqy3i3gqbk1z36ngil0k-emacs-...")
  #<subr delete-file>("/gnu/store/61zhv116kl3waqy3i3gqbk1z36ngil0k-emacs-..." nil)
  apply(#<subr delete-file> ("/gnu/store/61zhv116kl3waqy3i3gqbk1z36ngil0k-emacs-..." nil))
  delete-file("/gnu/store/61zhv116kl3waqy3i3gqbk1z36ngil0k-emacs-..." nil)
  package-recompile(org-roam)
  funcall-interactively(package-recompile org-roam)
  command-execute(package-recompile record)
  execute-extended-command(nil "package-recompile" "package-recompile")
  funcall-interactively(execute-extended-command nil "package-recompile" "package-recompile")
  command-execute(execute-extended-command)
```

Any clues or pointers?

**Comment by emfrom** [comment]
It looks like you are trying to recompile the compiled binaries that are in store, which is and should be read only. In other words you have at least roam installed via guix which means this solution wont work for you since emacs is not managing the org-roam package in your case.

Possible solutions:
1. Removing the org-roam package from guix and installing with package, use-package or straight inside emacs
2. Pull org and roam via git and prepend their loadpaths in emacs
3. Make sure you have the latest versions or org and org-roam in your guix environment (aka that they arent locked to some version)
4. Update the versions of org and org in the guix package database (yourself or nag a maintainer) 

Hope that helps.

**Comment by rolandog** [comment]
Thank you @emfrom for your much appreciated response and guidance.

Following your suggestion, I was actually able to resolve the issue by uninstalling & reinstalling org-roam with guix itself!  Everything is working properly now.

```sh
# for anyone else that stumbles into the same issue
guix package --remove emacs-org-roam
guix package --install emacs-org-roam
```

**Comment by elondres-mim** [comment]
In case this was useful to anyone else I had this problem while defining a custom dailies template, and removing my customization (to use the default) fixed the issue.

I'm on emacs-plus on macOS.

**Comment by rolandog** [comment]
Thanks for that @elondres-mim ; I'll also point to what worked for me (using Emacs+org-roam with Guix as a package manager on a foreign distro); i.e. manually load a couple of elisp files from their location in `/gnu/store/`: https://github.com/org-roam/org-roam/issues/2516#issuecomment-2930561986
